# Upstream: https://gitlab.gnome.org/GNOME/gnome-settings-daemon/merge_requests/57
#
From 74354d37b9e7cba1512513d1e55a538ed962c28a Mon Sep 17 00:00:00 2001
From: Ethan Hsieh <ethan.hsieh@canonical.com>
Date: Tue, 9 Oct 2018 10:28:31 +0800
Subject: [PATCH] rfkill: set encoding as NULL

The default encoding for GIOChannel is UTF-8, but rfkill event is binary
data. If the value is invalid UTF-8, gsd-rfkill-manager will fail to
receive rfkill event.
---
 plugins/rfkill/rfkill-glib.c | 1 +
 1 file changed, 1 insertion(+)

Index: gnome-settings-daemon-3.28.1/plugins/rfkill/rfkill-glib.c
===================================================================
--- gnome-settings-daemon-3.28.1.orig/plugins/rfkill/rfkill-glib.c
+++ gnome-settings-daemon-3.28.1/plugins/rfkill/rfkill-glib.c
@@ -426,6 +426,7 @@ cc_rfkill_glib_open (CcRfkillGlib  *rfki
 
 	/* Setup monitoring */
 	rfkill->channel = g_io_channel_unix_new (fd);
+	g_io_channel_set_encoding (rfkill->channel, NULL, NULL);
 	rfkill->watch_id = g_io_add_watch (rfkill->channel,
 					   G_IO_IN | G_IO_HUP | G_IO_ERR,
 					   (GIOFunc) event_cb,
